users/views.py

```diff
@@ -70,46 +71,144 @@ def user_logout(request):
 
 
 @login_required(login_url="login")
-def main_page(request):
-    cal = get_calandar_days()
-    current_month = get_current_month_name()
-    current_date = (datetime.now().month, datetime.now().day)
+def update_user(request):
+    user = request.user
+    form = UserUpdateForm(instance=user)
+
+    if request.method == "POST":
+        form = UserUpdateForm(request.POST, request.FILES, instance=user)
+
+        if form.is_valid():
+            form.save()
+            return redirect('home')
+
+    return render(request, "update_profile/update_form.html", {'form': form})
+
+
+@login_required(login_url="login")
+def main_page(request, date=None):
+
+    if not date:
+        year, month = None, None
+    else:
+        year, month = date.split("|")
+        year = int(year)
+        month = int(month)
+
+    cal = get_calendar_days(year, month)
+    current_month = get_current_month_name(month)
+    current_date = (datetime.now().year, datetime.now().month, datetime.now().day)
+    current_events = Event.objects.filter(eventparticipant__user_id=request.user.id)
+    table_measurements = get_table_measurements(current_events)
     week_days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+
     return render(
         request, "user_site/home.html",
-        {'week_days': week_days, 'calandar': cal, 'month_name': current_month, 'current_date': current_date}
+        {'week_days': week_days, 'calendar': cal, 'month_name': current_month,
+         'next_date': get_dates(year, month, True), 'previous_date': get_dates(year, month, False),
+         'max_height': table_measurements[0], 'max_margin': table_measurements[1],
+         'max_margin_events': table_measurements[2], 'max_padding': table_measurements[3],
+         'current_date': current_date, 'current_events': current_events}
     )
 
 
+def get_dates(year=None, month=None, next=True):
+    """ return the next or previous date.
+        next: if true the next date will be returned otherwise the previous
+    """
+
+    if not year:
+        year = int(datetime.now().year)
+    if not month:
+        month = int(datetime.now().month)
+
+    if next:
+        action = 1
+    else:
+        action = -1
```

Comment:
you can do it in one line like:
```suggestion
    action = 1 if next else -1
```

Reply:
done. thanks 😄